File: apps/backend/src/modules/tenant/entities/company.entity.ts
import { Column, Entity, OneToMany } from 'typeorm';
import { BaseEntity } from '../../../shared/database/base.entity';
import { Site } from './site.entity';

@Entity({ name: 'companies' })
export class Company extends BaseEntity {
  @Column({ type: 'varchar', length: 100, unique: true })
  code!: string;

  @Column({ type: 'varchar', length: 255 })
  name!: string;

  @Column({ type: 'text', nullable: true })
  description?: string;

  @Column({ type: 'varchar', length: 255, nullable: true, name: 'logo_url' })
  logoUrl?: string;

  @Column({ type: 'varchar', length: 100, nullable: true })
  timezone?: string;

  @Column({ type: 'varchar', length: 10, nullable: true })
  currency?: string;

  @Column({ type: 'boolean', default: true, name: 'is_active' })
  isActive!: boolean;

  @OneToMany(() => Site, (site) => site.company)
  sites!: Site[];
}

File: apps/frontend/lib/src/features/company/domain/entities/company_entity.dart
class CompanyEntity {
  const CompanyEntity({
    required this.id,
    required this.code,
    required this.name,
    this.description,
    this.logoUrl,
    this.timezone,
    this.currency,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String code;
  final String name;
  final String? description;
  final String? logoUrl;
  final String? timezone;
  final String? currency;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;
}

